docs/utils/threading/worker-pool.md

```diff
@@ -148,3 +148,217 @@ functionality in Thunder, which will be further described in a different documen
+        {
+            ASSERT(job.IsValid() == true);
+
+            bool revoked = (Revoke(job) != Core::ERROR_UNKNOWN_KEY);
+            Schedule(time, job);
+            return (revoked);
+        }
+```
+
+
+#### Revoking job
+`Revoke()` is used to cancel the execution of a `job`. We use it by specifying the `job` we want to cancel and and waiting time for `job` cancellation. We do not need to specify the time, if we leave the default value, the method will wait indefinitely until `job` is cancelled or an error occurs. The `Revoke()` method returns a numeric value.
+
+ Error codes that can be returned:
+
+* `ERROR_NONE` - 0 (no errors)
+* `ERROR_UNKNOWN_KEY` - 22
+
+!!!warning
+    `ERROR_UNKNOWN_KEY` will be returned in case the job was not previously scheduled!
+
+```cpp
+uint32_t Revoke(const Core::ProxyType<IDispatch>& job, const uint32_t waitTime = Core::infinite) override
+        {
+            uint32_t result(_timer.Revoke(Timer(this, job)) ? Core::ERROR_NONE : Core::ERROR_UNKNOWN_KEY);
+
+            uint32_t report = _threadPool.Revoke(job, waitTime);
+
+            if (report == Core::ERROR_UNKNOWN_KEY) {
+                report = _external.Completed(job, waitTime);
+
+                if ( (report != Core::ERROR_UNKNOWN_KEY) && (result == Core::ERROR_UNKNOWN_KEY) ) {
+                    result = report;
+                }
+            }
+
+            return (result);
+        }
+```
+#### Example
+
+Below is test example of using it.
+```cpp
+void test_scheduler(Core::ProxyType<IDispatch> job) {
+    // Schedule the job to run 5 seconds from now
+    Core::Time scheduledTime = Core::Time::Now() + Core::TimeSpan::FromSeconds(5);
+    _workerpool.Schedule(scheduledTime, job);
+
+    // Wait for a while to see the scheduled job execution
+    Core::Thread::Sleep(Core::TimeSpan::FromSeconds(10));
+
+    // Reschedule the job to run 10 seconds from now
+    Core::Time rescheduledTime = Core::Time::Now() + Core::TimeSpan::FromSeconds(10);
+    _workerpool.Reschedule(rescheduledTime, job);
+
+    // Wait for a while to see the rescheduled job execution
+    Core::Thread::Sleep(Core::TimeSpan::FromSeconds(15));
+
+    // Revoke the job (cancel its execution)
+    _workerpool.Revoke(job);
+}
+```
+For more examples you can check `Thunder/Tests/unit/core/test_workerpool.cpp` file.
+
+
+### Scheduler class
+`Scheduler` is a small class designed to schedule jobs in the future. It has one public method `Schedule()`. It takes as arguments the `job` whose execution is to be scheduled in the future and the time after which the execution of the `job` should start.
+```cpp
+void Schedule(const Time& time, const ProxyType<IDispatch>& job) override {
+                _timer.Schedule(time, Timer(_pool, job));
+            }
+```
+The Scheduler uses the `Timer` class of the `Workerpool` class as a privileged class. How the `Timer` class is built and how it works can be seen in the `Timer.h` file.
```

Comment:
1. I'm not sure I understand this properly - _"The Scheduler uses the `Timer` class of the `Workerpool` class as a privileged class."_ - maybe we could change it to _"The Scheduler uses a private `Workerpool` class called `Timer`."_ - does this carry the same meaning or have you meant something else?
2. _"How the `Timer` class is built and how it works can be seen in the `Timer.h` file."_ - it looks to me that in Timer.h we have a templated TimerType class, so we should be more exact here - mention that Timer (the private class of Workerpool) is used as a templated parameter of TimerType, which can be found in Timer.h etc.